File: 2 Add Two Numbers/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    dummyHead := &ListNode{0, nil};
    current := dummyHead;
    carry := 0;

    for (l1 != nil || l2 != nil || carry != 0) {
        v1 := 0
		if l1 != nil { v1 = l1.Val }
        v2 := 0
		if l2 != nil { v2 = l2.Val }
        sum := carry + v1 + v2;
        carry = int(sum / 10);
        current.Next = &ListNode{sum % 10, nil};
        current = current.Next;
		if l1 != nil { l1 = l1.Next }
		if l2 != nil { l2 = l2.Next }
    }

    return dummyHead.Next;
}

File: 2 Add Two Numbers/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        dummyHead = ListNode(0)
        current = dummyHead
        carry = 0

        while (l1 != None or l2 != None or carry != 0):
            v1 = l1.val if l1 != None else 0
            v2 = l2.val if l2 != None else 0
            sum = carry + v1 + v2
            carry = int(sum / 10)
            current.next = ListNode(sum % 10)
            current = current.next
            l1 = l1.next if l1 != None else None
            l2 = l2.next if l2 != None else None
        
        return dummyHead.next